### PR TITLE
Move CI scripts to docker runnable locally

### PR DESCRIPTION
**Pull request by kamilglod**

### CODE REVIEW

File: bin/tests.sh [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+cd "`dirname $0`"
+cd ..
+
+PG_IMAGE=postgres:${PG_VERSION:-"12"}-alpine
+PG_CONTAINER_NAME=postgres
+PYTHON_VERSION=${PYTHON_VERSION:-"3.8"}
+DOCKERFILE=dev-${PYTHON_VERSION}.Dockerfile
+
+echo "starting tests for POSTGRESQL ${PG_VERSION}"
+sed "s/^FROM python:.*$/FROM python:$PYTHON_VERSION/g" dev.Dockerfile | tee $DOCKERFILE
```
Comment (abahdanovich) [inline bin/tests.sh]: sad :(

https://docs.docker.com/engine/reference/builder/#understand-how-arg-and-from-interact

> FROM instructions support variables that are declared by any ARG instructions that occur before the first FROM.

gdyby nie działało, to jest jeszcze standardowy toolik `envsubst` do tworzenia plików z templatki (korzystałem, działa)

Comment (kamilglod) [reply bin/tests.sh]: indeed, it works :)

File: README.md [hunk 2]
```diff
@@ -51,4 +51,17 @@ conn.close()
...
+* date
+
+
+### Development:
+```bash
+docker-compose -f dev.docker-compose.yaml run dev bash
+pip install --user -e .[dev,e2e]
+pytest tests/
+```
+If you need to test different PostgreSQL and Python version, you can locally modify dev.docker-compose.yaml or use CI script:
```
Comment (abahdanovich) [inline README.md]: i don't really like the idea of having 2 right ways of running tests.. in my opinion we should:

1. have full-featured local test runner (e.g. using docker-compose)
2. make CI server use follow out way to run tests (https://docs.travis-ci.com/user/docker/#using-docker-compose)

File: .travis.yml [hunk 3]
```diff
@@ -18,19 +18,14 @@ jobs:
   include:
     - stage: test
       env:
-        - PG_VERSION=10
-        - PYTHON_VERSION=3.7
+        - PG_VERSION=9
+        - PYTHON_VERSION=3.8
```
[the remark this answers is not in the document]
Comment (kamilglod) [reply .travis.yml]: PG12 is tested with both 3.7 and 3.8. Other postgres versions are tested only with python3.8 as I believe that there is no point of testing whole PGxPython matrix

File: bin/tests.sh [hunk 4]
```diff
@@ -6,20 +6,21 @@ set -o pipefail
...
+PG_IMAGE=postgres:${PG_VERSION}-alpine
 PG_CONTAINER_NAME=postgres
 PYTHON_VERSION=${PYTHON_VERSION:-"3.8"}
-DOCKERFILE=dev-${PYTHON_VERSION}.Dockerfile
 
-echo "starting tests for POSTGRESQL ${PG_VERSION}"
-sed "s/^FROM python:.*$/FROM python:$PYTHON_VERSION/g" dev.Dockerfile | tee $DOCKERFILE
+echo "starting tests for POSTGRESQL ${PG_VERSION} and PYTHON ${PYTHON_VERSION}"
 
+mkdir ./.pip_cache || echo
```
[the remark this answers is not in the document]
Comment (kamilglod) [reply bin/tests.sh]: internal docker volumes does not have explicit host path which makes it much harder to use inside CI (right now it is used by travis to cache `.pip_cache` dir between builds)

Using venv inside container does not have much more benefits when compared with user scoped installs but it requires extra setup steps.

however I understand benefits that comes from using virtualenv inside docker. I could try to manage it somehow.

File: bin/deploy.sh [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+cd ..
+
+export UID
+docker-compose down
+docker-compose config
+docker-compose build
+docker-compose run \
+  -e TWINE_USERNAME=${TWINE_USERNAME} \
+  -e TWINE_PASSWORD=${TWINE_PASSWORD} \
+  dev bash -c "
```
Comment (abahdanovich) [inline bin/deploy.sh]: ERROR: No such service: dev

**Pull request merged**